pkg/context/fake/fake_vm_context.go

```diff
@@ -17,33 +17,35 @@ limitations under the License.
 package fake
 
 import (
+	"context"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/cluster-api/util/patch"
 
 	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
-	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
+	capvcontext "sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
 )
 
 // NewVMContext returns a fake VMContext for unit testing
 // reconcilers with a fake client.
-func NewVMContext(ctx *context.ControllerContext) *context.VMContext {
+func NewVMContext(controllerCtx *capvcontext.ControllerContext) *capvcontext.VMContext {
```

Comment:
Same here, let's pass a context and use it instead of context.Background.

```suggestion
func NewVMContext(ctx context.Context, controllerCtx *capvcontext.ControllerContext) *capvcontext.VMContext {
```

Reply:
Done